libbeat/common/bus/bus.go

```diff
@@ -0,0 +1,102 @@
+package bus
+
+import (
+	"sync"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+// Event sent to the bus
+type Event common.MapStr
+
+// Bus provides a common channel to emit and listen for Events
+type Bus interface {
+	// Emit an event to the bus
+	Emit(Event)
```

Comment:
I think `Publish` would be a more expressive name than `Emit`, as it is a kind of Publish-Subscribe model.